### PR TITLE
Registry 2.7 release and arm binaries for registry 2.6

### PR DESCRIPTION
**Pull request by caervs**

### CODE REVIEW

**Comment by caervs** [comment]
cc @dmcgowan @davidswu

**Comment by caervs** [comment]
@tianon bump 😸

**Comment by lag-linaro** [comment]
LGTM

**Comment by yosifkit** [comment]
As far as I can tell we are still waiting on https://github.com/docker/distribution-library-image/pull/83 and https://github.com/docker/distribution-library-image/pull/84 since Alpine 3.4 doesn't support multi-architecture (only [amd64](https://github.com/docker-library/official-images/blob/dda7a9d5805eb2cdc46a660f5622536d14838fc2/library/alpine#L111-L112)).

**Comment by caervs** [comment]
Cool. Thanks for looking this over. https://github.com/docker/distribution-library-image/pull/83 and https://github.com/docker/distribution-library-image/pull/84 just got merged.

File: library/registry [hunk 1]
```diff
@@ -2,9 +2,21 @@ Maintainers: David Wu (@davidswu),
...
+Architectures: amd64, arm64v8, arm32v6
+GitFetch: refs/heads/registry-v2.7.0
+GitCommit: c8fb369af7ecefd44047f5140a5180ac8d643100
+amd64-Directory: amd64
+arm32v6-Directory: arm
+arm64v8-Directory: arm64
+
+Tags: 2.6.2, 2.6
+Architectures: amd64, arm64v8, arm32v6
+GitFetch: refs/heads/registry-2.6
```
Comment (yosifkit) [inline library/registry]: I don't see a `registry-2.6` branch in docker/distribution-library-image but I do see a `release/2.6` branch that was updated with https://github.com/docker/distribution-library-image/pull/83.

And then the GitCommit will need to be updated (probably `2798058c4eadcbb2510df72d50f01e59f393480b`, but the merge commit is also fine https://github.com/docker/distribution-library-image/commit/fc40f1f1051bb4a42ee4661ccaa190c1bd6c6be9).

**Comment by caervs** [comment]
@yosifkit comments addressed 😺

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 4060877..0e18bcd 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,2 +1,7 @@
 registry:2.5 @ amd64
+registry:2.6 @ amd64
+registry:2.6 @ arm32v6
+registry:2.6 @ arm64v8
 registry:latest @ amd64
+registry:latest @ arm32v6
+registry:latest @ arm64v8
diff --git a/_bashbrew-list b/_bashbrew-list
index a345918..f3a9732 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -3,4 +3,6 @@ registry:2.5
 registry:2.5.2
 registry:2.6
 registry:2.6.2
+registry:2.7
+registry:2.7.0
 registry:latest
diff --git a/registry_latest/Dockerfile b/registry_2.6/Dockerfile
similarity index 68%
copy from registry_latest/Dockerfile
copy to registry_2.6/Dockerfile
index 7ef40c6..27199b4 100644
--- a/registry_latest/Dockerfile
+++ b/registry_2.6/Dockerfile
@@ -1,12 +1,12 @@
 # Build a minimal distribution container
 
-FROM alpine:3.4
+FROM alpine:3.8
 
 RUN set -ex \
     && apk add --no-cache ca-certificates apache2-utils
 
-COPY ./registry/registry /bin/registry
-COPY ./registry/config-example.yml /etc/docker/registry/config.yml
+COPY ./registry /bin/registry
+COPY ./config-example.yml /etc/docker/registry/config.yml
 
 VOLUME ["/var/lib/registry"]
 EXPOSE 5000
diff --git a/registry_latest/registry/config-example.yml b/registry_2.6/config-example.yml
similarity index 100%
rename from registry_latest/registry/config-example.yml
rename to registry_2.6/config-example.yml
diff --git a/registry_2.5/docker-entrypoint.sh b/registry_2.6/docker-entrypoint.sh
similarity index 100%
copy from registry_2.5/docker-entrypoint.sh
copy to registry_2.6/docker-entrypoint.sh
diff --git a/registry_latest/registry/registry b/registry_2.6/registry
similarity index 75%
copy from registry_latest/registry/registry
copy to registry_2.6/registry
index 9a80175..6181809 100755
Binary files a/registry_latest/registry/registry and b/registry_2.6/registry differ
diff --git a/registry_latest/Dockerfile b/registry_latest/Dockerfile
index 7ef40c6..27199b4 100644
--- a/registry_latest/Dockerfile
+++ b/registry_latest/Dockerfile
@@ -1,12 +1,12 @@
 # Build a minimal distribution container
 
-FROM alpine:3.4
+FROM alpine:3.8
 
 RUN set -ex \
     && apk add --no-cache ca-certificates apache2-utils
 
-COPY ./registry/registry /bin/registry
-COPY ./registry/config-example.yml /etc/docker/registry/config.yml
+COPY ./registry /bin/registry
+COPY ./config-example.yml /etc/docker/registry/config.yml
 
 VOLUME ["/var/lib/registry"]
 EXPOSE 5000
diff --git a/registry_2.5/registry/config-example.yml b/registry_latest/config-example.yml
similarity index 100%
copy from registry_2.5/registry/config-example.yml
copy to registry_latest/config-example.yml
diff --git a/registry_latest/registry/registry b/registry_latest/registry
similarity index 58%
rename from registry_latest/registry/registry
rename to registry_latest/registry
index 9a80175..3557b59 100755
Binary files a/registry_latest/registry/registry and b/registry_latest/registry differ
```

</details>

**Comment by yosifkit** [comment]
Build test of #5141; 386382d8; `amd64` (`registry`):

```console

$ bashbrew build registry:2.7.0
Building bashbrew/cache:8809356cd7ad6222d5a28214a0482d70bae7a5050a73e6fb124e4a52c60f4964 (registry:2.7.0)
Tagging registry:2.7.0
Tagging registry:2.7
Tagging registry:2
Tagging registry:latest

$ test/run.sh registry:2.7.0
testing registry:2.7.0
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'docker-registry-push-pull' [5/5]...passed


$ bashbrew build registry:2.6.2
Building bashbrew/cache:ee25ab904ba29ed63f80485c056973af51076a8d5c9706598b6b6eaa61e8dd18 (registry:2.6.2)
Tagging registry:2.6.2
Tagging registry:2.6

$ test/run.sh registry:2.6.2
testing registry:2.6.2
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'docker-registry-push-pull' [5/5]...passed


$ bashbrew build registry:2.5.2
Using bashbrew/cache:0e140347a8a824893b6700eb9d4936d3928f7af18aa2de29a647f1f7fc33fa19 (registry:2.5.2)
Tagging registry:2.5.2
Tagging registry:2.5

$ test/run.sh registry:2.5.2
testing registry:2.5.2
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'docker-registry-push-pull' [5/5]...passed

```

**Comment by yosifkit** [comment]
Build test of #5141; 386382d8; `arm64v8` (`registry`):

```console

$ bashbrew build registry:2.7.0
Building bashbrew/cache:bc25d3d557c3b850289b5eeec450b61226486174f4347de9e10a27995e3f002c (registry:2.7.0)
Tagging registry:2.7.0
Tagging registry:2.7
Tagging registry:2
Tagging registry:latest

$ test/run.sh registry:2.7.0
testing registry:2.7.0
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'docker-registry-push-pull' [5/5]...Unable to find image 'docker:dind' locally
dind: Pulling from library/docker
e3c488b39803: Already exists
05a63128803b: Already exists
4fe3f66f9bc2: Already exists
590d7dc3d210: Already exists
6997ca5468f9: Already exists
cc81cc08dc7c: Already exists
2f4b2c55ff3b: Already exists
f52e68576a3f: Pulling fs layer
51119cd31818: Pulling fs layer
23a6cdd926c1: Pulling fs layer
73396b657d75: Pulling fs layer
73396b657d75: Waiting
51119cd31818: Verifying Checksum
51119cd31818: Download complete
23a6cdd926c1: Verifying Checksum
23a6cdd926c1: Download complete
73396b657d75: Verifying Checksum
73396b657d75: Download complete
f52e68576a3f: Verifying Checksum
f52e68576a3f: Download complete
f52e68576a3f: Pull complete
51119cd31818: Pull complete
23a6cdd926c1: Pull complete
73396b657d75: Pull complete
Digest: sha256:df4199a7907517b54914b66ad6fe04a6250f4f6f3eb6f7cf8a836e856d1c1621
Status: Downloaded newer image for docker:dind
passed


$ bashbrew build registry:2.6.2
Building bashbrew/cache:4993ba6976837fe24980c4a99a045f17bbbe0b762dcc648c5bf2205b7483dfa7 (registry:2.6.2)
Tagging registry:2.6.2
Tagging registry:2.6

$ test/run.sh registry:2.6.2
testing registry:2.6.2
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'docker-registry-push-pull' [5/5]...passed


$ bashbrew build registry:2.5.2
skipping "registry:2.5.2" (due to architecture "arm64v8"; only "amd64" supported)

$ test/run.sh registry:2.5.2
testing registry:2.5.2
	image does not exist!

```


The following images failed at least one test:

- `registry`:
  - `registry:2.5.2`

**Pull request merged**